### ISSUE TITLE
[CinnVIIStarkMenu@NikoKrause] Unexpected/Inconvenient search results ordering

### ISSUE THREAD

**Issue by dzek69** [linuxmint/cinnamon-spices-applets#1556]
```
 * Applet version/Build date: 2017-11-16 15:45:15 UTC
 * Cinnamon version: 3.4.6
 * Distribution - Mint 18.2
 * Graphics hardware *and* driver used: nVidia GTX950M, nvidia-384.90-0ubuntu0.16.04.2
 ```

**Notify author of applet**
@NikoKrause

**Issue**
When I'm searching for something - The list contains items that has matching app name or app description. However - there is no importance set whenever app name or description is the important one and this causes unexpected results being on the first/upper place, like presented in screenshot:
![https://i.imgur.com/E3BBtfe.jpg](https://i.imgur.com/E3BBtfe.jpg)

**Steps to reproduce**
Install Skype and Telegram, make sure Skype has description like in screenshot. Search for `tele`.

OR (example that doesn't require installing something):

Add to the menu items like that:
"MyApplication" with description "Test description"
"Test application" (description doesn't matter)

Search for "test".

**Expected behaviour**
(Continuation from steps to reproduce)
Telegram (or "Test application") should be displayed first, because there is app name match here and Skype (or "MyApplication") should be second, because there is only a description match.

In short: results that has matching name should be presented before results with matching description.

**Other information**
None.

**Comment by NikoKrause** [comment]
This menu uses the same search algorithm as the default Cinnamon menu.
This means search results are not rearranged by relevance, but they are sorted alphabetically.

It's been an issue in Cinnamons menu for a long time, see: https://github.com/linuxmint/Cinnamon/issues/4053

I think it's done this way, because sorting menu entries costs a lot of time.

As soon as Cinnamon improves it's sorting algorithm, I can backport it. But for now I'll stay with the current solution. I'm afraid you have to type in `teleg` instead of `tele` for now. Or use the arrow keys.

I'm closing this as an upstream issue.

**Comment by NikoKrause** [comment]
This issue has been adressed upstream and CinnVIIStarkMenu version 1.29 includes the backported improvements.
